### ISSUE TITLE
Tool for Documentation

### ISSUE THREAD

**Issue by hassiahk** [enochkan/torch-metrics#13]
Hello Together,

What tool do we want to use for documentation? I know of some tools we can use:
- `Sphinx` (used by `PyTorch`)
- `DocTest` (used by `TensorFlow`)
- `MkDocs`
- `Markdown` files for each metric.

Personally, I would prefer Markdown files since they are easy to use. Let me know your thoughts @enochkan @zutshianand @Vijayabhaskar96

**Comment by Vijayabhaskar96** [comment]
I don't know about tools, but seriously need documentations. I began to write tests and couldn't figure out what is the input and shape, most simply says "Tensor of Predicted values." that is so vague, we need shape and clear info about input and outputs. Only when going through code I found currently Precision, Recall, and F1 are calculated only for binary classification problems.

**Comment by hassiahk** [comment]
Yes, need to change `Precision`, `Recall` and `F1`. Even `Accuracy` was designed to work only for binary classification, I changed it. I will try to include the shapes as well in the docstrings, thanks for letting me know.

**Comment by Vijayabhaskar96** [comment]
Thanks for doing this, also consider adding info on whether the function takes raw logits or thresholded predictions or works on both because It's not clear to me to write test cases right now. For example currently precision takes only floating-point inputs breaks when using int targets, but ints should also be fine the user is most likely to provide actual and predicted labels.

**Comment by hassiahk** [comment]
Let me know which of the following would be a good practice:
- Including examples of all the ways you can use a metric in the docstring (just code) and also documentation (will have more details than just code).
- Including a sample example for a metric in docstring and various examples in the documentation.

**Comment by Vijayabhaskar96** [comment]
I prefer the 1st option. But also, libraries like keras(standalone) autogenerate docs from function docstrings using tools, so docstrings contain the whole documentation for the function. Chose which will be the easier way going further.

**Comment by hassiahk** [comment]
Yes, that's why I wanted to know which tool we want to use for documentation. :smile:

**Comment by enochkan** [comment]
Let's stick to Markdown files for each metric for now. Can refactor in the future if needed. @hassiahk do you have the bandwidth to work on this?

**Comment by hassiahk** [comment]
> @hassiahk do you have the bandwidth to work on this?

Not for next two to three weeks. I can only work on this after that.

**Comment by aiqc** [comment]
check out sphinx with nbsphinx for notebooks

live example: https://aiqc.readthedocs.io/en/latest/notebooks/keras_multi-label_classification.html

readthedocs is listening to my github repo via gtihub webhooks
